compiler/src/lntoamm/Type.ts

```diff
@@ -502,25 +502,23 @@ export class Type {
     const otherTypeAst = fulltypenameAstFromString(otherType.typename)
     let generics = []
     if (typeAst.has('opttypegenerics')) {
-      const genericsAst = typeAst.get('opttypegenerics').get('generics')
-      generics.push(genericsAst.get('fulltypename').t)
-      genericsAst.get('cdr').getAll().forEach(r => {
+      generics.push(typeAst.get('opttypegenerics').get('generics').get('fulltypename').t)
+      typeAst.get('opttypegenerics').get('generics').get('cdr').getAll().forEach(r => {
         generics.push(r.get('fulltypename').t)
       })
     }
     generics = generics.map(
-      g => scope.deepGet(g) || Type.fromStringWithMap(g, interfaceMap, scope) as Type
+      g => scope.deepGet(g) || Type.fromStringWithMap(g, interfaceMap, scope) as Type || new Type('-bogus-', false, true)
```

Comment:
Instead of the `typeApplies` call crashing, I realized that it should always return a totally bogus type that shouldn't match with anything if it can't find the type at all. Then I got much saner errrors instead of stacktraces.